isthmus/src/test/java/io/substrait/isthmus/ExpressionConvertabilityTest.java

```diff
@@ -44,40 +58,50 @@ public void mapLiteral() throws IOException, SqlParseException {
   }
 
   @Test
-  public void singleOrList() throws IOException {
-    Plan.Root root =
-        b.root(
-            b.filter(
-                input -> b.singleOrList(b.fieldReference(input, 0), b.i32(5), b.i32(10)),
-                commonTable));
-    var relNode = converter.convert(root.getInput());
-    var expression =
-        ((Filter) relNode)
-            .getCondition()
-            .accept(
-                new RexExpressionConverter(
-                    CREATE_SEARCH_CONV.apply(relNode.getCluster().getRexBuilder()),
-                    new ScalarFunctionConverter(
-                        SimpleExtension.loadDefaults().scalarFunctions(), typeFactory)));
-    var to = new ExpressionProtoConverter(new ExtensionCollector(), null);
+  public void singleOrList() {
+    Expression singleOrList = b.singleOrList(b.fieldReference(commonTable, 0), b.i32(5), b.i32(10));
+    RexNode rexNode = singleOrList.accept(converter);
+    Expression substraitExpression =
+        rexNode.accept(
+            new RexExpressionConverter(
+                CREATE_SEARCH_CONV.apply(rexBuilder),
+                new ScalarFunctionConverter(extensions.scalarFunctions(), typeFactory)));
+
+    // cannot roundtrip test singleOrList because Calcite simplifies the representation
+    assertExpressionEquality(
+        b.or(
+            b.equal(b.fieldReference(commonTable, 0), b.i32(5)),
+            b.equal(b.fieldReference(commonTable, 0), b.i32(10))),
+        substraitExpression);
+  }
+
+  @Test
+  public void switchExpression() {
+    Expression switchExpression =
+        b.switchExpression(
+            b.fieldReference(commonTable, 0),
+            List.of(b.switchClause(b.i32(5), b.i32(1)), b.switchClause(b.i32(10), b.i32(2))),
+            b.i32(3));
+    RexNode rexNode = switchExpression.accept(converter);
+    Expression expression =
+        rexNode.accept(
+            new RexExpressionConverter(
+                CASE, new ScalarFunctionConverter(extensions.scalarFunctions(), typeFactory)));
+
+      // cannot roundtrip test switchExpression because Calcite simplifies the representation
+      assertExpressionEquality(
+        b.ifThen(
+            List.of(
+                b.ifClause(b.equal(b.fieldReference(commonTable, 0), b.i32(5)), b.i32(1)),
+                b.ifClause(b.equal(b.fieldReference(commonTable, 0), b.i32(10)), b.i32(2))),
+            b.i32(3)),
+        expression);
+  }
```

Comment:
These tests were creating a full plan and converting it, but we only actually want/need to verify the _expression_ conversion.

Removed the plan related noise by converting the expression directly. Also used a bunch of our nice utils (including one I added) to make the tests more readable.